src/lib/datepicker/datepicker-input.ts

```diff
@@ -358,15 +360,16 @@ export class MatDatepickerInput<D> implements ControlValueAccessor, OnDestroy, V
 
   _onInput(value: string) {
     let date = this._dateAdapter.parse(value, this._dateFormats.parse.dateInput);
+    this._lastValueValid = !date || this._dateAdapter.isValid(date);
     const current = this._selectionModel.getFirstSelectedDate();
     date = this._getValidDateOrNull(date);
 
-    if (!this._dateAdapter.sameDate(current, date) && date) {
+    if (!this._dateAdapter.sameDate(current, date)) {
       this._selectionModel.add(date, true);
       this._cvaOnChange(date);
       this.dateInput.emit(new MatDatepickerInputEvent(this, this._elementRef.nativeElement));
     } else {
-      this._validatorOnChange();
+      this._cvaOnChange(date);
```

Comment:
this might need to be changed back. Thought all tests pass, not sure about the differences